### PR TITLE
only stream audio when necessary

### PR DESCRIPTION
**Pull request by sehraf**
Here is a small patch that stop streaming audio when idle and hotword detection is set to local.

### CODE REVIEW

**Review by Romkabouter — COMMENTED** [review]
Only the xEventGroupSetBits should be in the if statement, otherwise the semaphore does not get freed. initHeader does not matter

Also, this should be implemented in Idle state as well

File: PlatformIO/src/StateMachine.hpp [hunk 1]
```diff
@@ -68,14 +68,27 @@ class Idle : public StateMachine
     if (xSemaphoreTake(wbSemaphore, (TickType_t)10000) == pdTRUE) {
       device->updateColors(COLORS_IDLE);
     }
-    xEventGroupSetBits(audioGroup, STREAM);
-    xSemaphoreGive(wbSemaphore);
-    initHeader(device->readSize, device->width, device->rate);
+    // start streaming audio to the remote endpoint for hotword detection
```
Comment (Romkabouter) [inline PlatformIO/src/StateMachine.hpp:71]: Sorry, already implement I see. Same issue as in HotwordDetected, only the GroupSetBits should be in the if statement.

File: PlatformIO/src/StateMachine.hpp [hunk 2]
```diff
@@ -68,14 +68,27 @@ class Idle : public StateMachine
...
+      xEventGroupSetBits(audioGroup, STREAM);
+      xSemaphoreGive(wbSemaphore);
+      initHeader(device->readSize, device->width, device->rate);
+    }
   }
 
   void run(void) override {
     if (device->isHotwordDetected() && !hotwordDetected) {
       hotwordDetected = true;
+
```
Comment (Romkabouter) [inline PlatformIO/src/StateMachine.hpp:83]: This not needed, the device starts a new session and then enters HotwordDetected state which starts the stream.
It should not be started here.

**Comment by Romkabouter** [comment]
Thanks for the PR! I have a couple remarks, can you check them?

**Comment by sehraf** [comment]
force pushed. Thanks for the comments, i wasn't aware that the dialogmanager triggers streaming anyway.

**Pull request merged**